packages/cache/src/stylis-plugins.js

```diff
@@ -1,89 +1,100 @@
-// @flow
-// https://github.com/thysultan/stylis.js/tree/master/plugins/rule-sheet
-// inlined to avoid umd wrapper and peerDep warnings/installing stylis
-// since we use stylis after closure compiler
+import { tokenize } from '@emotion/stylis'
 
-import type { StylisPlugin } from './types'
+const last = arr => (arr.length ? arr[arr.length - 1] : null)
 
-const delimiter = '/*|*/'
-const needle = delimiter + '}'
+export let compat = element => {
+  switch (element.type) {
+    case 'rule':
+      var i = 0
+      var props = element.props
+      var value = element.value
+      var propsLength = props.length
 
-function toSheet(block) {
-  if (block) {
-    Sheet.current.insert(block + '}')
-  }
-}
+      // short-circuit for the simplest case
+      if (propsLength === 1 && value.charCodeAt(0) !== 58 /* colon */) {
+        return
+      }
 
-export let Sheet: { current: { +insert: string => void } } = {
-  current: (null: any)
-}
+      var valueTokens = tokenize(value)
+      // keep delimiting indices between tokens of a particular group
+      // include -1 and the length as well to allow for uniform calculations later
+      var points = [-1]
 
-export let ruleSheet: StylisPlugin = (
-  context,
-  content,
-  selectors,
-  parents,
-  line,
-  column,
-  length,
-  ns,
-  depth,
-  at
-) => {
-  switch (context) {
-    // property
-    case 1: {
-      switch (content.charCodeAt(0)) {
-        case 64: {
-          // @import
-          Sheet.current.insert(content + ';')
-          return ''
-        }
-        // charcode for l
-        case 108: {
-          // charcode for b
-          // this ignores label
-          if (content.charCodeAt(2) === 98) {
-            return ''
-          }
+      for (; i < valueTokens.length; i++) {
+        if (valueTokens[i].charCodeAt(0) === 44 /* , */) {
```

Comment:
Commas in css functions i.e `:not(h1,h2)`, aren't handled here, you might be better of just re-passing `element.value` to `compile` and extract  it i.e `var props = compile(element.value+'{}')[0].props` if you are re-parsing it all over again here.

```js
compile(element.value+'{}')[0].props.map(value => {
if (value.charAt(0) === ':')
return // handle your psuedo selector
})
```

I've come up with a short function for this

```js
var compat = function ({type, value, props} => {
if (type === 'rule')
	if (~value.indexOf(':'))
		for (var i = 0; i < props.length; i++);
		    props[i] = props[i].replace(' :', ':')
})
```

But it doesn't yet handle the `.xyz{& first-child{}}` case.

Reply:
> Commas in css functions i.e :not(h1,h2)

Are u sure? parens are actually grouped by `tokenize`, so we get this:
```js
tokenize('.foo :not(h1,h2)') // [".foo", " ", ":", "not", "(h1,h2)"]
```
so I don't encounter a comma token at all for this case - a comma in a css function won't be accounted as a "point" (rightfully so).

> you might be better of just re-passing element.value to compile and extract it i.e var props = compile(element.value+'{}')[0].props 

as far as I can tell this cannot be done so easily because:
- i can't just override props like this as original props have already full computed selectors and value holds only the "current level" stuff, so doing it like this i would remove selector parts from the "parent levels" (those from lower depths of the input)
- i still have no notion of `&` is at the "current level" and need to extract it from the props anyhow

> I've come up with a short function for this compat plugin

This doesn't handle `div,:focus` case - a colon doesn't have to be first in the value, it has to be first in the "group" (`:focus` being a second group here)

It would also fail on `:first-child,div :last-child{}` as it would remove the space from the second group.

> But it doesn't yet handle the .xyz{& first-child{}} case.

Well, that's exactly that pesky case I've tried to handle through all of this overcomplicated stuff 😅